### PR TITLE
Add image and requirements_file params to black task

### PR DESCRIPTION
**Pull request by frerikandriessen**
This provides more flexibility to the user for which exact version of
black should be used. 

# Changes
You can now specify which image to use, and which requirement_file to use. Aside from allowing you to specify the black version, this will also allow you to more closely mimic your eventual production environment.
See also #792 for a similar PR. 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [X] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [X] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [X] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [X] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [X] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [X] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: task/black/0.2/README.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Black (Python Code Prettier)
+
+This task can be used to format the python source code using [Black](https://github.com/psf/black) which is an Opinionate Code Formatter.
+
+## Installing the Task
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/black/0.1/black.yaml
```
Comment (vinamra28) [inline task/black/0.2/README.md:8]: nit
```suggestion
kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/black/0.2/black.yaml
```

**Review by vinamra28 — COMMENTED** [review]
just one minor fix else and after that it can be merged

File: task/black/0.2/README.md (hunk 1 above)
Comment (frerikandriessen) [reply task/black/0.2/README.md]: Done

**Review by vinamra28 — APPROVED** [review]
Thank you for adding the PR
/meow

**Comment by PuneetPunamiya** [comment]
Thanks for adding
/lgtm

**Pull request merged**